utilities/generate_messages_catalog.py

```diff
@@ -131,96 +134,169 @@ def _is_callable(item):
         return False
 
 
-def _build_id_messages(python_package, package_root):
+def _find_modules_and_directories(top_level_directory):
+    """
+    Recursive helper function to find all python files included in top level
+    package. This will recurse down the directory paths of any package to find
+    all modules and subpackages in order to create an exhaustive list of all
+    python files within a given package.
+
+    :param top_level_directory: Path to the top level of a python package.
+    :type top_level_directory: str
+    :return: Returns a list of paths to all python files within that package.
+    :rtype: list [str]
+    """
+    modules = []
+    directories = []
+
+    for item in os.listdir(top_level_directory):
+        if item.endswith(".py"):
+            modules.append(os.path.join(top_level_directory, item))
+        elif os.path.isdir(os.path.join(top_level_directory, item)):
```

Comment:
This is a little confusing, it never actually adds the directories. It adds directories to their own list to be recursed and added to the module list, so it will only ever add python files. If that path had no python files, the sub-folders might still have them as there are not `__inits__.py` at every level unfortunately. If/when we get that corrected in the future, we could enhance this one to not add a directory if no python files were, short-circuiting some of the looping, but for now, since this is run offline during the build process, the extra wasted loops is not a huge concern.